builders/pallets-precompiles/precompiles/gmp.md

```diff
@@ -0,0 +1,155 @@
+---
+
+# Interacting with the GMP Precompile
+
+![GMP Precompile Moonbeam Banner](/images/builders/pallets-precompiles/precompiles/gmp/gmp-banner.png)
+
+## Introduction {: #introduction } 
+
+Moonbeam Routed Liquidity refers to Moonbeam’s use case as the port parachain for liquidity into other Polkadot parachains through a combination of chain-agnostic cross-chain protocols and XCMP. 
+
+The GMP precompile acts as an interface for Moonbeam Routed Liquidity, acting as a middleman between bridging protocols and parachains connected to Moonbeam via XCM. Currently the GMP precompile only supports the relaying of liquidity through Wormhole.
+
+The GMP Precompile is located at the following address:  
+
+=== "Moonbase Alpha"
+     ```
+     {{networks.moonbase.precompiles.gmp}}
+     ```
+
+## The GMP Solidity Interface {: #the-gmp-solidity-interface }
+
+[`Gmp.sol`](https://github.com/PureStake/moonbeam/blob/master/precompiles/gmp/Gmp.sol){target=_blank} is a Solidity interface that allows developers to interact with the precompile’s methods:  
+
+- **wormholeTransferERC20**(*bytes memory* vaa) - receives a wormhole bridge transfer VAA, mints tokens via the wormhole token bridge, and forwards the liquidity to the custom payload’s MultiLocation
+
+In practice, it is unlikely that a developer will have to directly interact with the precompile. Relayers interact with the precompile to complete cross-chain actions, so the origin chain that the cross-chain action originates is where the developer has the most responsibility.
+
+The most common instance that someone will have to interact with the precompile is in the case of a recovery, where a relayer doesn’t complete an MRL transaction. A user would have to find the VAA that comes with their origin chain transaction and then manually invoke the wormholeTransferERC20 function.
+
+## Building the Payload for Wormhole
+
+Currently the GMP precompile only supports sending liquidity through Wormhole.  
+
+To send liquidity, on the origin chain, users must invoke the [`transferTokensWithPayload` method](https://book.wormhole.com/technical/evm/tokenLayer.html#contract-controlled-transfer){target=_blank} on the [WormholeTokenBridge smart contract](https://github.com/wormhole-foundation/wormhole/blob/main/ethereum/contracts/bridge/interfaces/ITokenBridge.sol){target=_blank}. It requires a bytes payload, which must be formatted as a SCALE encoded MultiLocation object wrapped within another versioned type.  
+
+You may be unfamiliar with both SCALE encoding and MultiLocations if you are not familiar with the Polkadot ecosystem. [SCALE encoding](https://docs.substrate.io/reference/scale-codec/){target=_blank} is a compact form of encoding that Polkadot uses. The [MultiLocation type](https://wiki.polkadot.network/docs/learn-xcvm){target=_blank} is used to define a relative point in Polkadot, such as a specific account on a specific parachain (Polkadot blockchain).  
```

Comment:
We don't define multilocation in any of the other precompile interfaces, but to compensate I'm adding a link to a multilocation definition in the general XCM location. Do you think that would be enough for a reader? Most developers will likely just b-line it to the Wormhole section anyways

Reply:
yeah the other XCM precompiles are a bit different since they're just sections embedded into pages which include the definition.

But now that you have pointed it out, I think this is something we can probably improve on. Maybe create an XCM-specific (or interoperability-specific) glossary page so it's easier to link to these definitions... 

Thanks for bringing it up! And yeah, your solution is good, thanks!